api/client/retl/sources.go

```diff
@@ -0,0 +1,108 @@
+package retl
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// CreateRetlSource creates a new RETL source
+func (r *RudderRETLStore) CreateRetlSource(ctx context.Context, source *RETLSourceCreateRequest) (*RETLSource, error) {
+	data, err := json.Marshal(source)
+	if err != nil {
+		return nil, fmt.Errorf("marshalling source: %w", err)
+	}
+
+	resp, err := r.client.Do(ctx, "POST", "/retl-sources", bytes.NewReader(data))
```

Comment:
I made a change in my recent PR for the Account API client, where I'm not adding the /v2 prefix by default on all paths. The reason is that there was a bug when pagination was used (it is introduced in accounts APIs) where the /v2 was also added to the `next` url that server was responding, which already included v2. I also think it's better if we explicitly add the version on the endpoints in case we want to introduce other versions in the future.

That PR is now merged, so let's rebase this one on main, and add "/v2/" in front of those paths.